src/toil/batchSystems/lsf.py

```diff
@@ -24,274 +24,171 @@
+            process = subprocess.Popen(
+                    ["bjobs", "-o", "jobid stat start_time delimiter='|'"],
+                    stdout=subprocess.PIPE)
+            stdout, _ = process.communicate()
+
+            for curline in stdout.split('\n'):
+                items = curline.strip().split('|')
+                if items[0] in currentjobs and items[1] == 'RUN':
+                    jobstart = parse(items[2], default=datetime.now(tzlocal()))
+                    times[currentjobs[items[0]]] = datetime.now(tzlocal()) \
+                        - jobstart
+            return times
+
+        def killJob(self, jobID):
+            subprocess.check_call(['bkill', self.getBatchSystemID(jobID)])
+
+        def prepareSubmission(self, cpu, memory, jobID, command):
+            return self.prepareBsub(cpu, memory, jobID) + [command]
+
+        def submitJob(self, subLine):
+            combinedEnv = self.boss.environment
+            combinedEnv.update(os.environ)
+            process = subprocess.Popen(subLine, stdout=subprocess.PIPE,
+                                       env=combinedEnv)
+            line = process.stdout.readline()
+            logger.debug("BSUB: " + line)
+            result = int(line.strip().split()[1].strip('<>'))
+            logger.debug("Got the job id: {}".format(result))
+            return result
+
+        def getJobExitCode(self, lsfJobID):
+            # the task is set as part of the job ID if using getBatchSystemID()
+            job, task = (lsfJobID, None)
+            if '.' in lsfJobID:
+                job, task = lsfJobID.split('.', 1)
+
+            # first try bjobs to find out job state
+            args = ["bjobs", "-l", str(job)]
+            logger.debug("Checking job exit code for job via bjobs: "
+                         "{}".format(job))
+            process = subprocess.Popen(args, stdout=subprocess.PIPE,
+                                       stderr=subprocess.STDOUT)
+            started = 0
+            for line in process.stdout:
+                if "Done successfully" in line:
+                    logger.debug("bjobs detected job completed for job: "
+                                 "{}".format(job))
+                    return 0
+                elif "New job is waiting for scheduling" in line:
+                    logger.debug("bjobs detected job pending scheduling for "
+                                 "job: {}".format(job))
+                    return None
+                elif "PENDING REASONS" in line:
+                    logger.debug("bjobs detected job pending for job: "
+                                 "{}".format(job))
+                    return None
+                elif "Exited with exit code" in line:
+                    exit = int(line[line.find("Exited with exit code ")+22:]
+                               .split('.')[0])
+                    logger.debug("bjobs detected job exit code: "
+                                 "{}".format(exit))
```

Comment:
We were never reporting the actual exit code of the underlying process, which is shocking.